library/std/src/process/tests.rs

```diff
@@ -417,6 +417,41 @@ fn env_empty() {
     assert!(p.is_ok());
 }
 
+#[test]
+#[cfg(not(windows))]
+#[cfg_attr(any(target_os = "emscripten", target_env = "sgx"), ignore)]
+fn main() {
+    const PIDFD: &'static str =
+        if cfg!(target_os = "linux") { ", create_pidfd: false" } else { "" };
+
+    let mut command = Command::new("some-boring-name");
+
+    assert_eq!(
+        format!("{command:?}"),
+        format!(
+            r#"Command {{ program: "some-boring-name", args: ["some-boring-name"], env: CommandEnv {{ clear: false, saw_path: false, vars: {{}} }}, cwd: None, uid: None, gid: None, saw_nul: false, groups: None, stdin: None, stdout: None, stderr: None, pgroup: None{PIDFD} }}"#
```

Comment:
I personally would prefer if the debug output still resembled a commandline invocation. I often copy-paste the rustc invocation cargo and rustbuild show directly into the terminal when debugging. These are generated by the debug impl I believe. For the program name, arguments and env vars the debug output I prefer should be obvious. For eg cwd you could prefix `cd the/cwd &&`. For stdio you could use eg `<stdin_file >stdout_file 2>stderr_file`. For the rest you could maybe do something like `@uid=0`. `saw_nul` is an internal field I believe and shouldn't be printed.

Reply:
I think a command-like display would better be used by implementing `Display` rather than change the `Debug` one

Reply:
There is no Display impl right now and it would take a long time before cargo would be able to adapt it. In the mean time copy-paste would be broken.

Reply:
One's should never rely on debug information, it has no guarantee of stability. This is not an issue with the implementation but rather with the original issue itself, which as been triaged as valid.

Reply:
> One's should never rely on debug information, it has no guarantee of stability.

Sure, I'm not parsing it, so changes are not breaking anything. It is just that changes to `Command`'s `Debug` impl may make manual actions, like copy-pasting to execute the command again, more cumbersome.

Reply:
True. I'm not sure how common it is that the command fails when run in a rust program but works when entered manually *and* is what the user wants.

If it's common we could leave the command like output at the end, ready to be copied?

Reply:
although it would not seem very clean

Reply:
If you invoke `./x.py` with enough `-v` arguments, it will list the exact commands used to invoke rustc using this debug impl. This is useful for running in a debugger or slightly changing the command while debugging. It also shows it when a rustc command failed when only using a single `-v`: https://github.com/rust-lang/rust/blob/a2cdcb3fea2baae5d20eabaa412e0d2f5b98c318/src/bootstrap/bin/rustc.rs#L230

Reply:
I'm just worried the use case is too specific to include it in the debug implementation

Reply:
To me it is **the** main use case of `impl Debug for Command {}`.

Reply:
I've reintroduced the command-like output after the struct-like output
```
Command { program: "some-boring-name", args: ["exciting-name", "1", "2", "3"], env: CommandEnv { clear: false, saw_path: false, vars: {} }, cwd: None, uid: None, gid: None, saw_nul: false, groups: None, stdin: None, stdout: None, stderr: None, pgroup: None{PIDFD} }
Command line: ["some-boring-name"] "exciting-name" "1" "2" "3"
```